### ISSUE TITLE
App should not crash at start if one of the ZIM file is corrupted

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-desktop#1128]
I have achieved to get a corrupted ZIM file, see https://github.com/kiwix/kiwix-desktop/issues/1127 for the details.

Trying to open the file, Kiwix just crashes. Here the details:
https://tmp.kiwix.org/corrupted_zim_crash/

When I restart Kiwix, it crash at start and I can not use Kiwix at all anymore

What I expect:
If there is crash by reading a ZIM file, I expect Kiwix to detect it and stop (somehow) crashing when I restart it.

**Comment by veloman-yunkan** [comment]
@kelson42 Can you check if the `kiwix-desktop` crash was fixed by openzim/libzim#895?

**Comment by kelson42** [comment]
@veloman-yunkan Indeed, I have retested again.
```
$ ./kiwix-desktop
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
add widget
terminate called after throwing an instance of 'zim::ZimFileFormatError'
  what():  Error parsing cluster. Offsets are not ordered.
Aborted (core dumped)
```

But it seems that this time this is because the exception is not catched (properly).

Maybe you can fix this at the same time?
